src/Authenticator/FormAuthenticator.php

```diff
@@ -50,26 +49,43 @@ class FormAuthenticator extends AbstractAuthenticator
     ];
 
     /**
-     * Constructor
+     * Authenticates the identity contained in a request. Will use the `config.userModel`, and `config.fields`
+     * to find POST data that is used to find a matching record in the `config.userModel`. Will return false if
+     * there is no post data, either username or password is missing, or if the scope conditions have not been met.
      *
-     * @param \Authentication\Identifier\IdentifierInterface $identifier Identifier or identifiers collection.
-     * @param array<string, mixed> $config Configuration settings.
+     * @param \Psr\Http\Message\ServerRequestInterface $request The request that contains login information.
+     * @return \Authentication\Authenticator\ResultInterface
      */
-    public function __construct(IdentifierInterface $identifier, array $config = [])
+    public function authenticate(ServerRequestInterface $request): ResultInterface
```

Comment:
Is there a reason we can't use `setIdentifier()`/`getIdentifier()`? It seems it could be a simpler path forward.

If when we need the identifier, and the collection is empty, authenticators could create their default. We could even have a hook method on `AbstractAuthenticator` if you wanted to get fancy.

Reply:
> If when we need the identifier, and the collection is empty, authenticators could create their default.

That's what's being done below isn't it? Just that the `_identifier` property is used directly instead of using `setIdentifer()`/`getIdentifier()`.

Reply:
Do you want to make an alternative PR or adjust this one?
Or all good now?